buffer/src/main/java/io/netty/buffer/api/Buffer.java

```diff
@@ -212,6 +212,14 @@
      */
     boolean readOnly();
 
+    /**
+     * Queries if this buffer is backed by native memory, or not.
+     *
+     * @return {@code true} if this buffer is backed by native, off-heap, memory. Otherwise, {@code false}, if this
+     * buffer is backed by on-heap memory.
```

Comment:
@chrisvest I  just wonder if `isDirect()` could be false and it is still non on-heap in theory. Like for example an empty buffer.

Reply:
Empty buffers are not special in the new API. You can allocate empty off-heap buffers, and empty on-heap buffers.